qa/tasks/deepsea_deploy.py

```diff
@@ -716,6 +724,47 @@ def _run_stage_2(self, config):
         self._dump_global_conf()
         self._dump_mon_conf()
 
+    def _custom_ceph_conf(self, config):
+        """
+        Add custom options for ceph.conf
+        Dict definition should be defined between stage 2 and 3.
+        Conf files for each roles are stored in:
+        /srv/salt/ceph/configuration/files/ceph.conf.d/
+        EXAMPLE:
+        tasks:
+          - deepsea_deploy:
+              cli: true
+              commands:
+                - stage0:
+                    update: false
+                - stage1:
+                - policy_cfg:
+                    profile: default
+                - stage2:
+                - custom_ceph_conf:
+                    global:
+                      mon lease: 15
+                      mon lease ack timeout: 25
+                    mon:
+                      debug mon: 20
+                    osd:
+                      debug filestore: 20
+                - stage3:
+        """
+        log.info("deepsea_deploy: WWWW: custom options in ceph.conf")
+        if not config:
+            info_msg = ("deepsea_deploy: empty config ")
+            log.info(info_msg)
+            return
+        for section in config.keys():
+            if config[section] is not None:
```

Comment:
Change this conditional to `if isinstance(config[section], dict):`

("not None" will be true for things that aren't iterable, and in that case the next line would produce an unhandled exception)

Reply:
great, changed